packages/@react-types/shared/src/inputs.d.ts

```diff
@@ -41,7 +41,11 @@ export interface ValueBase<T, C = T> {
 }
 
 export interface TextInputBase {
-  /** Temporary text that occupies the text input when it is empty. */
+  /**
+   * Temporary text that occupies the text input when it is empty.
+   * Please use help text instead.
+   * @deprecated
+   **/
```

Comment:
This deprecation message shows up in the aria useTextField hook: 
<img width="741" alt="image" src="https://user-images.githubusercontent.com/9661127/161172445-c7bf988d-6a65-4b47-9ddc-7faa27bb282c.png">

Is it possible to omit this version of the placeholder in such a way that the `placeholder` type from `TextInputDOMProps` shows up in the aria hook props and this one shows up for the spectrum components? May wanna double check whether or not we wanna deprecate `placeholder` for the aria hooks in the first place though.

If we do want to deprecate `placeholder` for the aria hooks as well, we should probable drop the "Please use help text instead" since that is Spectrum specific.

Reply:
Opened https://github.com/adobe/react-spectrum/pull/3006 as a potential solution. Separated into its own PR so its a bit easier to look at. Definitely open to hearing better solutions, and we can merge that into this when its ready.

Reply:
Does repositioning the order of `TextInputDOMEvents` vs the base prop help for the aria prop case? e.g. `export interface AriaTextFieldProps extends TextFieldProps, AriaLabelingProps, FocusableDOMProps, **TextInputDOMProps**, AriaValidationProps {` -> `export interface AriaTextFieldProps extends **TextInputDOMProps**, TextFieldProps, AriaLabelingProps, FocusableDOMProps, AriaValidationProps {`. Then the Spectrum prop could re-extend the TextInputProps to get the deprecated placeholder prop again.

Either that or maybe using Omits to make sure the aria props always get the `TextInputDOMEvents`'s placeholder prop?

This approach might be more jankey than what you have in #3006. Also is dependent on `TextInputDomEvent` always being used along side `TextInputProps`

Reply:
Ended up using a `SpectrumTextInputBase` from https://github.com/adobe/react-spectrum/pull/3006 to deprecate placeholder, which should be a bit less confusing and more explicit than needing to re-order and re-extend anything. Double checked, and all the types show placeholder to be deprecated where expected.